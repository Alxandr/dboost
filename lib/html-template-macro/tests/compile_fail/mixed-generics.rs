use html_template::{HtmlAttributeValue, HtmlTemplate};
use html_template_macro::component;

component! {
	/// This is some comments
	#[derive(Debug)]
	pub(crate) struct TestComponent<T>(
		title: T,
		attribute: impl HtmlAttributeValue + HtmlTemplate,
		children: impl HtmlTemplate
	)
	where
		T: Into<String>,
	{
		<main />
	}
}

fn main() -> fmt::Result {
	unreachable!()
}
