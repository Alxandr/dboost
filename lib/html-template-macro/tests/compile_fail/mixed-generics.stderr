error: Cannot use impl traits *and* explicit generics at the same time
 --> tests/compile_fail/mixed-generics.rs:9:14
  |
9 |         attribute: impl HtmlAttributeValue + HtmlTemplate,
  |                    ^^^^

error: Cannot use impl traits *and* explicit generics at the same time
  --> tests/compile_fail/mixed-generics.rs:10:13
   |
10 |         children: impl HtmlTemplate
   |                   ^^^^

error: Cannot use impl traits *and* explicit generics in one component.
 --> tests/compile_fail/mixed-generics.rs:7:33
  |
7 |     pub(crate) struct TestComponent<T>(
  |                                    ^

error[E0433]: failed to resolve: use of undeclared crate or module `fmt`
  --> tests/compile_fail/mixed-generics.rs:19:14
   |
19 | fn main() -> fmt::Result {
   |              ^^^ use of undeclared crate or module `fmt`
